routes/userProfile-alt.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2014-2021 Bjoern Kimminich.
+ * SPDX-License-Identifier: MIT
+ */
+
+const fs = require('fs')
+const models = require('../models/index')
+const utils = require('../lib/utils')
+const insecurity = require('../lib/insecurity')
+const challenges = require('../data/datacache').challenges
+const pug = require('pug')
+const config = require('config')
+const themes = require('../views/themes/themes').themes
+
+module.exports = function getUserProfile () {
+  return (req, res, next) => {
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg" width="20"/></sub>&nbsp; Allocation of Resources Without Limits or Throttling

This endpoint handler performs a file system operation and does not use a rate-limiting mechanism. It may enable the attackers to perform Denial-of-service attacks. Consider using a rate-limiting middleware such as express-limit.

<h6>
<a href="https://cwe.mitre.org/data/definitions/770.html">CWE-770</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 522</a> | Line 16</h6>